Marlin/src/feature/tmc_util.cpp

```diff
@@ -905,9 +915,11 @@
     TMC_REPORT("Stallguard thrs",    TMC_SGT);
 
     DRV_REPORT("DRVSTATUS",          TMC_DRV_CODES);
+    #if HAS_TMCX1X0 || HAS_TMC220x
+      DRV_REPORT("sg_result",        TMC_SG_RESULT);
+    #endif
     #if HAS_TMCX1X0
```

Comment:
This would be a good place for those macros I mentioned, such as `HAS_STALLGUARD`.

I didn't rework all of these because I was being extremely cautious not to change behavior. If you understand it well enough to know you could apply a more intuitively named macro, go for it!

Reply:
My goal with those `HAS_` macros was to consolidate knowledge about which features are supported by each driver model into one place, rather than scattering it throughout the codebase.

Reply:
From the change it looks like you can have a report for some drivers that would include `TMC_SG_RESULT` but not `TMC_STALLGUARD`. Is this an error, or are there some drivers that actually do report `TMC_SG_RESULT` but not `TMC_STALLGUARD`?

Reply:
TMC_STALLGUARD is wired to bit 24 on DRV_STATUS which is a single bit signaling whether the stall threshold was hit and it is not implemented in all drivers(not in 2209 for instance)
TMC_SG_RESULT is a separate 10 bit register in some drivers(2209) or 10 bits of DRV_STATUS in other drivers(such as 2130). Clear as mud?

Reply:
At least that clear. I'm making a matrix with all the TMC drivers on the Y and all the features on the X. From that I can make a Google Sheet where you can check off the features you want and it will show the drivers that have them. Hopefully the final result will not hurt the eyes too much.